notebooks/API.py

```
@@ -0,0 +1,189 @@
+encoder = joblib.load('/home/nazar/Muhammad Raees Azam/JazzCash_Raees/Code/jazzcash/models/lgbm_encoder.pkl')
+
+# Business rules configuration
+high_risk_cities = [101, 202]  # Example high-risk cities
+high_risk_states = [10, 12]  # Example high-risk states
+new_account_threshold = 10000  # Account ID threshold for newly created accounts
+amount_threshold = 100000  # Transaction amount threshold
+n, m = 3, 5  # Rule 1: 3 transactions in 5 minutes
+
+# Define input data model using Pydantic
+class TransactionInput(BaseModel):
+    Transaction_Id: Optional[str] = Field(alias="Transaction_Id")
```

Ruff (0.8.2)

Undefined name `Optional`

(F821)